window/src/os/windows/window.rs

```diff
@@ -1093,6 +1279,53 @@ fn apply_mouse_cursor(cursor: Option<MouseCursor>) {
     }
 }
 
+unsafe fn check_title_rmouse(hwnd: HWND, msg: UINT, screen_coords: ScreenPoint) -> Option<LRESULT> {
```

Comment:
For https://github.com/wez/wezterm/issues/1485 I've been thinking of teaching the wezterm-gui layer how to produce its own context menu windows (I don't *really* want to do my own, but the native context menus generally block the rest of the GUI until a selection is made which poses some implementation challenges on various OSs).

In that world, the choice to show and what to show are all handled in the wezterm-gui layer rather than in the window layer.

My semi-formed thoughts around this:

* The title bar area has some ideal locations for the minimize, maximize and close buttons (on the RHS on Windows, on the LHS on macOS, and not prevailing choice on linux), so it would make sense for the window crate to expose positioning information to the application layer so that the fancy tab bar would make space for them and/or have a way to trigger rendering them.  That doesn't need to happen here in this PR, I'm just sharing what I've been thinking.
* The decision about when to show a context menu should really be done in the application layer, which knows where its UIItems reside
* We could add a `WindowOps::show_windows_system_context_menu` function that does the `GetSystemMenu`/`TrackPopupMenuEx` bit of this function, and then teach `mouse_event_tab_bar` to call it for right clicks on `TabBarItem::None` as a way to get this functionality working in the meantime, and then ultimately replace it with the full blown context menu feature once that is done

Reply:
Good points. I'll implement like your suggestion from the last 2 items.

As for the first one, do you mean to render native window control buttons? I'm not sure how that would work on other OSes, but on Windows there's no API to make the system draw the window control buttons, you have to draw them yourself. It's funny, because after knowing that, I started noticing that indeed every app that has a custom title bar does exactly that. They try to make it similar to the native Windows 10 one, but each one is slightly different and have slightly different behaviors. 😂

There's actually a new way, but it's only available for UWP/Windows 11 apps.